### PR TITLE
Add create_repository_dispatch

### LINKED ISSUE #1299: Support for repository dispatch event
Preview feature useful for triggering github actions.

POST /repos/:owner/:repo/dispatches
https://developer.github.com/v3/repos/#create-a-repository-dispatch-event

### PR DESCRIPTION
**Pull request by christopher-dG**
Closes #1299 

The code is working fine but I'm not quite sure what kind of tests I should be adding or where. Some guidance there would be very much appreciated!

### CODE REVIEW

File: github/Repository.py [hunk 1]
```diff
@@ -1373,6 +1373,28 @@ def __create_pull(self, **kwds):
...
+            client_payload, dict
+        ), client_payload
+        post_parameters = {"event_type": event_type}
+        if client_payload is not github.GithubObject.NotSet:
+            post_parameters["client_payload"] = client_payload
+
+        headers, data = self._requester.requestJsonAndCheck(
+            "POST", self.url + "/dispatches", input=post_parameters
+        )
+        return None
```
Comment (s-t-e-v-e-n-k) [inline github/Repository.py]: This returns a 204, and we already return a truthy value in that case. See other places in this very file for "return status == 204".

File: github/Repository.py [hunk 2]
```diff
@@ -1373,6 +1373,28 @@ def __create_pull(self, **kwds):
...
         )
 
+    def create_repository_dispatch(
+        self, event_type, client_payload=github.GithubObject.NotSet
+    ):
+        """
+        :calls: POST /repos/:owner/:repo/dispatches <https://developer.github.com/v3/repos/#create-a-repository-dispatch-event>
+        :param event_type: string
+        :param client_payload: dict
+        :rtype None
```
Comment (s-t-e-v-e-n-k) [inline github/Repository.py]: I think this is missing a : ?

Comment (s-t-e-v-e-n-k) [reply github/Repository.py]: And should be updated when you change the code too, of course.

**Comment by christopher-dG** [comment]
Added a test, I think I did it right.

**Pull request merged**